pkg/k8s/policyhelper/policy.go

```diff
@@ -2,27 +2,64 @@ package policyhelper
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"strings"
 
 	"golang.org/x/exp/slices"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
 	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
 )
 
+var (
+	ErrGroupKind         = errors.New("group/kind error")
```

Comment:
"group/kind error" is not indicative of the issue. Is the problem that the user is attempting to create a resource with the wrong group/kind? If so, perhaps "invalid group/kind" is more descriptive.

Reply:
this error for error type checking, actual descriptive error below in the code

https://github.com/aws/aws-application-networking-k8s/blob/af8774450d0a8c698c0d82945e81081a66c8b4e1/pkg/k8s/policyhelper/policy.go#L371-L374